presto-main/src/main/java/com/facebook/presto/sql/planner/SqlPlannerContext.java

```diff
@@ -73,64 +69,23 @@ public class CteInfo
         @VisibleForTesting
         public static final String delimiter = "_*%$_";
         // never decreases
-        private int currentQueryScopeId;
-
-        // Maps a set of Query objects, including the parent query statement and all its referenced statements,
-        // to a unique scope identifier. Each set of related queries shares the same scope.
-        Map<TreeSet<Query>, String> queryNodeScopeIdMap = new HashMap<>();
-
-        public String normalize(Analysis analysis, Query query, String cteName)
-        {
-            QueryReferenceCollectorContext context = new QueryReferenceCollectorContext();
-            context.getReferencedQuerySet().add(query);
-            query.accept(new QueryReferenceCollector(analysis), context);
-            TreeSet<Query> normalizedKey = context.getReferencedQuerySet();
-            if (!queryNodeScopeIdMap.containsKey(normalizedKey)) {
-                queryNodeScopeIdMap.put(normalizedKey, String.valueOf(currentQueryScopeId++));
-            }
-            return queryNodeScopeIdMap.get(normalizedKey) + delimiter + cteName;
-        }
-
-        private class QueryReferenceCollector
-                extends DefaultTraversalVisitor<Void, QueryReferenceCollectorContext>
-        {
-            private final Analysis analysis;
+        private int prefix;
 
-            public QueryReferenceCollector(Analysis analysis)
-            {
-                this.analysis = analysis;
-            }
-
-            @Override
-            protected Void visitTable(Table node, QueryReferenceCollectorContext context)
-            {
-                Analysis.NamedQuery namedQuery = analysis.getNamedQuery(node);
-                if (namedQuery != null) {
-                    context.addQuery(namedQuery.getQuery());
-                    process(namedQuery.getQuery(), context);
-                }
-                return null;
-            }
-        }
+        // Map a cte Query to a unique ID, which will be used in CTE reference node to identify the same CTE
+        BiMap<NodeRef<Query>, String> cteQueryUniqueIdMap = HashBiMap.create();
 
-        private class QueryReferenceCollectorContext
+        public String normalize(NodeRef<Query> queryNodeRef, String cteName)
         {
-            private final TreeSet<Query> referencedQuerySet;
-
-            public QueryReferenceCollectorContext()
-            {
-                this.referencedQuerySet = new TreeSet<>(Comparator.comparingInt(Query::hashCode));
-            }
-
-            public void addQuery(Query ref)
-            {
-                this.referencedQuerySet.add(ref);
+            if (cteQueryUniqueIdMap.containsKey(queryNodeRef)) {
+                return cteQueryUniqueIdMap.get(queryNodeRef);
             }
-
-            public TreeSet<Query> getReferencedQuerySet()
-            {
-                return referencedQuerySet;
+            String identityString = queryNodeRef.hashCode() + delimiter + cteName;
```

Comment:
Use hash code to uniquely identify a CTE

Reply:
I am not sure if repeated runs of the query will have the same hashcode.
If they do then please ignore.
If they don't, is it better to normalize this hashcode by ID too like done previously? 
The goal is to make debugging easier in prod since its better for repeated query runs to have the same id of the cte, also the test framework changes might not be required

Reply:
Good point. Instead of using hash code, now I use an incremental prefix for the mapping purpose, which should be fixed now, and get rid of the massive change of test framework.